### PR TITLE
[FLCRM-12860] Fix filter records of operator DateYesterday

### PR DESCRIPTION
**Pull request by dotconde**
https://fulcrumapp.atlassian.net/browse/FLCRM-12860

In Fulcrum, checkout to `bugfix/FLCRM-12860-check` branch, modify package.json so that `fulcrum-query-sql` points to the following hash commit

```
github:fulcrumapp/fulcrum-query-sql#866d86df6018f582372e4b7394f7b3845d1abab7
```

Delete the existing `yarn.lock` file, rebuild the lock file by running `yarn install`

Re-skaffold and the changes will now be visible for debugging

Videotesting:

From San Francisco (GMT -7)
https://github.com/fulcrumapp/fulcrum/assets/45726165/fd3c3e1e-b1cf-4963-8f5c-d5c859db6c85

From Tokyo (GMT +9)
https://github.com/fulcrumapp/fulcrum/assets/45726165/248fd24e-30d7-486c-b6fb-f4443efa5cf9

### CODE REVIEW

**Comment by dotconde** [comment]
This ticket involves two parts:

1. A change in fulcrum-query-sql (this PR).
2. A change in fulcrumapp to correctly send the date range when using the DateYesterday operator.
- The current issue arises for customers in non-US time zones, such as `Europe/London` or `Asia/Shanghai`.
- In these cases, the error persists because an incorrect date range is sent, for example (for August 28th):
`WHERE ("date" BETWEEN '2023-08-26T23:00:00.000Z' AND '2023-08-26T23:00:00.000Z')`.
- The proposed PR in fulcrum-query-sql is a good starting point to address this problem. However, a complete solution will require another ticket in fulcrumapp to correct how the date range is sent to the DateYesterday operator.

**Comment by dotconde** [comment]
https://github.com/fulcrumapp/fulcrum-query-sql/assets/45726165/690178e8-81c8-4651-98d8-457291f2e823


https://github.com/fulcrumapp/fulcrum-query-sql/assets/45726165/c00ae37a-c44c-4af5-b608-2d950dcb60da


https://github.com/fulcrumapp/fulcrum-query-sql/assets/45726165/4ac5b2ef-06b9-47d5-a088-e48ec5c6a8e7

File: src/operator.js [hunk 1]
```diff
@@ -621,13 +621,17 @@ export function availableOperatorsForColumn(column) {
...
-export function calculateDateRange(operator, value, now) {
+export function calculateDateRange(column, operator, value, now) {
   now = moment(now || new Date()).clone().startOf('day');
 
   const date1 = now.clone();
   const date2 = now.clone();
 
   const range = (start, end) => {
+    if (column.type === 'date') {
+      return [ start.clone(), (end || start).clone() ];
```
[the remark this answers is not in the document]
Comment (dotconde) [reply src/operator.js:632]: This modification was made in the initial approach using `DynamicDateConverter` and it did not completely solve the problem for all timezones, I just created a new converter and will probably remove this logic.

**Review by jacobpayne659 — APPROVED** [review]

**Review by toddrun — APPROVED** [review]
This seems sane to me (though to be honest, I did not do any testing).

**Pull request closed without merging**